src/rules/javascript/webextension-unsupported-api.js

```diff
@@ -28,6 +28,15 @@ const rule = {
             return;
           }
 
+          if (
+            privileged &&
+            addonMetadata?.experimentApiPaths?.some((apiPath) =>
+              api.startsWith(apiPath)
```

Comment:
given that we only care about the namespace, could we change this check to `apiPath === namespace`?
(and if we could do that, maybe we may opt to leave `experimentaApiPaths` to be of type Set and just do `addonMetadata?.experimentApiPaths?.has(module)`). 

I'm wondering also if by checking for `api.startsWith` we may be actually ignoring an API call that isn't technically defined by "experiment_apis"  but may still share a prefix with one that is defined and pass this check, but I haven't checked if that is actually technically possible.

Not a super big deal given that this is mainly silencing warnings and so even if there was some reported as a warning it wouldn't be blocking signing the addon.

Reply:
I don't think `apiPath === namespace` would work for compound api paths, e.g., `some.experiment`.

Reply:
Here is an example: https://github.com/mozilla-extensions/searchengine-devtools/blob/2726d3e07a6f9723eda370f8635f4fcc5df976c4/extension/manifest.json#L41

The namespace should be `experiments.searchengines` I think. Using `api` in the check works currently but that wouldn't work for `foo.bar.baz.api()`. We should fix how we compute the `namespace` but that does not seem trivial.

Reply:
See: https://github.com/mozilla/addons-linter/issues/4364